src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java

```diff
@@ -488,6 +490,23 @@ public void dumpAndRestore() {
     assertTrue(jedis.exists("foo2"));
   }
 
+  @Test(expected = JedisBusyException.class)
+  public void dumpAndRestoreDup() {
+    jedis.set("foo1", "bar1");
+    byte[] sv = jedis.dump("foo1");
+    jedis.restore("foo1", 0, sv);
+    assertTrue(jedis.exists("foo1"));
+  }
+
+  @Test
+  public void dumpAndRestoreReplace() {
+    jedis.set("foo1", "bar1");
+    byte[] sv = jedis.dump("foo1");
+    assertTrue(jedis.exists("foo1"));
+    jedis.restore("foo1", 0, sv, Protocol.DATA_REPLACE);
```

Comment:
i had considered about Params but , it looks like Params contains Key-value , replace is the only arg , i am not sure where the "REPLACE" should be placed.